docs/sources/k6/v1.0.x/extensions/create/guide.md

```diff
@@ -0,0 +1,371 @@
+
+In this guide, you’ll learn how to:
+
+- Create a GitHub repository using the k6 extension template repository.
+- Create a TypeScript declaration file to document your API.
+- Create [ascii85](https://en.wikipedia.org/wiki/Ascii85) encoding and decoding implementation.
+- Build a k6 binary with the extension.
+- Use the custom k6 binary to run a test.
+- Best practices for creating tests, checking for security vulnerabilities, and static analysis for your extension.
+
+For this guide, you’ll implement two functions that handle ascii85 encoding, which is a feature that’s not natively supported by k6. This will be implemented using Go.
+
+## Before you begin
+
+To follow along, you’ll need:
+
+- A [GitHub account](https://docs.github.com/en/get-started/start-your-journey/creating-an-account-on-github).
+
+Having a GitHub account simplifies the process of developing k6 extensions, which the guide will cover. [GitHub Codespaces](https://github.com/features/codespaces) provides a streamlined development experience for k6 extensions, reducing the need for local setup.
+
+## Create a GitHub repository
+
+The first step is to create a GitHub repository using the [grafana/xk6-example](https://github.com/grafana/xk6-example) template repository. This can be done interactively in a browser by clicking [here](https://github.com/new?template_name=xk6-example&template_owner=grafana). Name the repository "xk6-example-ascii85", and set the visibility to **Public**.
+
+Alternatively, use the [GitHub CLI](https://cli.github.com/) to create the repository.
+
+```bash
+gh repo create -p grafana/xk6-example -d "Experimental k6 extension" --public xk6-example-ascii85
+```
+
+## Create a codespace
+
+GitHub Codespaces is a GitHub feature that lets you create and use a fully configured development environment in the cloud.
+
+To create a GitHub codespace:
+
+- Go to the xk6-example-ascii85 repository you created in the previous step.
+- On the repository page, click the green **Code** button and then select **Codespaces** from the dropdown menu.
+- Click **Create new codespace**.
+
+Once the codespace is ready, it will open in your browser as a Visual Studio Code-like environment, letting you begin working on your project with the repository code already checked out.
+
+Alternatively, use the [GitHub CLI](https://cli.github.com/) to create the codespace, replacing `USER` with your GitHub username:
+
+```bash
+gh codespace create --repo USER/xk6-example-ascii85 --web
+```
+
+## Remove example code
+
+The xk6-example-ascii85 repository includes integration examples between Go and the k6 JavaScript runtime. These files can be useful as learning resources, but they’re not required for this extension.
+
+You can use the `.starter.patch` file with the `git apply` command to delete these files.
+
+```bash
+git apply .starter.patch
+rm .starter.patch
+```
+
```

Comment:
Please delete the "Remove example code" section. On the one hand, it can be confusing, and on the other hand, it can sometimes cause complications.